salt/lax/bot-lax-adaptor.sls

```diff
@@ -186,15 +186,10 @@ uwsgi-bot-lax-adaptor:
         - require:
             - service: uwsgi-bot-lax-adaptor
 
-    # test to ensure service is not serving up 500 responses
-    # the 'listen' statement ensures it runs at the end of a state run
-    # currently doesn't work: due to `listen` it runs 
-    # after the listener_nginx-server-service restart
-    # but it also runs immediately here, and since nginx is not restarted yet, it fails because no one is answering on port 8001 yet
-    #cmd.run:
-    #    - name: |
-    #        curl --silent --include --head --fail {{ apiprotocol }}://{{ apihost }}:8001/ui/
-    #    - listen:
-    #        - service: nginx
-    #        - service: uwsgi-bot-lax-adaptor
+uwsgi-bot-lax-smoke-test:
+    http.wait_for_successful_query:
+        - name: {{ apiprotocol }}://{{ apihost }}:8001/ui/
+        - status: 200
+        - wait_for: 10 # seconds. five checks with 1 second between each
+        - request_interval: 1 # second
```

Comment:
a `require` here would force the ordering of states, since it has been extracted from the previous state

Reply:
if this is a quality implementation it's going to be useful, I've been handrolling https://github.com/elifesciences/builder-base-formula/blob/master/elife/utils/wait_for_port in many places (it doesn't go into checking the status code, just the port)

Reply:
> a require here

oh yeah - can't believe I missed that